client/store/reports/stats/selectors.js

```diff
@@ -0,0 +1,52 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import { select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { ERROR } from 'store/constants';
+
+/**
+ * Returns report stats details for a specific endpoint query.
+ *
+ * @param  {Object} state     Current state
+ * @param  {String} endpoint  Stats endpoint
+ * @param  {Object} query     Report query paremters
+ * @return {Object}           Report details
+ */
+function getReportStats( state, endpoint, query ) {
+	const queries = get( state, [ 'reports', 'stats', endpoint ], {} );
+	const _query = query || {};
+	return queries[ JSON.stringify( _query, Object.keys( _query ).sort() ) ] || null;
```

Comment:
You can simplify these two lines (24-25) with
`return queries[ getQueryKey( query || {} ) ] || null;`